survey/src/main/java/com/quickbirdstudios/surveykit/backend/address/YandexAddressSuggestionProvider.kt

```diff
@@ -0,0 +1,85 @@
+package com.quickbirdstudios.surveykit.backend.address
+
+import com.quickbirdstudios.surveykit.AnswerFormat
+import java.io.FileNotFoundException
+import java.net.HttpURLConnection
+import java.net.URL
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.async
+import kotlinx.coroutines.launch
+import org.json.JSONArray
+import org.json.JSONException
+import org.json.JSONObject
+
+class YandexAddressSuggestionProvider(
+    private val apiKey: String,
+    private val resultsCount: Int = 5,
+    private val lang: String = "tr_TR",
+    onSuggestionListReady: (suggestions: List<AddressSuggestion>) -> Unit? = {}
+) : AddressSuggestionProvider(onSuggestionListReady) {
+
+    private fun getApiPath(apiKey: String, query: String, results: Int, lang: String) =
+        "https://geocode-maps.yandex.ru/1.x/?apikey=$apiKey" +
```

Comment:
What level of adress is `Yandex` being used? Exact address (Street, number etc)? Or just the city? If it is the second one, it can be done without yandex.

You can just use a `Geocoder` object for that purpose and extract properties like `adminArea` and `country` .